src/shared/discover-image.c

```diff
@@ -1590,6 +1590,21 @@ int image_set_limit(Image *i, uint64_t referenced_max) {
         return 0;
 }
 
+int image_set_pool_limit(uint64_t referenced_max) {
+        const char *dir = "/var/lib/machines";
```

Comment:
Maybe just make this an input argument to the function?

Reply:
Yeah. Please make this a generic function (i.e. call it `btrfs_set_dual_quota` or so) and place it under `btrfs-util`.

Reply:
instead of hardcoding /var/lib/machines, mght be a good idea to use image_root_to_string() and take an ImageClass param instead. i.e. one day we might want to reuse this for portabled and co, and it makes this less hard coded

Reply:
if you make this take a ImageClass, then please keep it in discover-image.c